packages/flutter_tools/lib/src/macos/native_assets_host.dart

```diff
@@ -13,55 +13,55 @@ import '../build_info.dart';
 import '../convert.dart';
 import '../globals.dart' as globals;
 
-/// The target location for native assets on macOS.
+/// Create an `Info.plist` in [target] for a framework with a single dylib.
 ///
-/// Because we need to have a multi-architecture solution for
-/// `flutter run --release`, we use `lipo` to combine all target architectures
-/// into a single file.
-///
-/// We need to set the install name so that it matches what the place it will
-/// be bundled in the final app.
-///
-/// Code signing is also done here, so that we don't have to worry about it
-/// in xcode_backend.dart and macos_assemble.sh.
-Future<void> copyNativeAssetsMacOSHost(
-  Uri buildUri,
-  Map<AssetPath, List<Asset>> assetTargetLocations,
-  String? codesignIdentity,
-  BuildMode buildMode,
-  FileSystem fileSystem,
+/// The framework must be named [name].framework and the dylib [name].
+Future<void> createInfoPlist(
+  String name,
+  Directory target,
 ) async {
-  if (assetTargetLocations.isNotEmpty) {
-    globals.logger.printTrace('Copying native assets to ${buildUri.toFilePath()}.');
-    final Directory buildDir = fileSystem.directory(buildUri.toFilePath());
-    if (!buildDir.existsSync()) {
-      buildDir.createSync(recursive: true);
-    }
-    for (final MapEntry<AssetPath, List<Asset>> assetMapping in assetTargetLocations.entries) {
-      final Uri target = (assetMapping.key as AssetAbsolutePath).uri;
-      final List<Uri> sources = <Uri>[for (final Asset source in assetMapping.value) (source.path as AssetAbsolutePath).uri];
-      final Uri targetUri = buildUri.resolveUri(target);
-      final String targetFullPath = targetUri.toFilePath();
-      await lipoDylibs(targetFullPath, sources);
-      await setInstallNameDylib(targetUri);
-      await codesignDylib(codesignIdentity, buildMode, targetFullPath);
-    }
-    globals.logger.printTrace('Copying native assets done.');
-  }
+
+  final File infoPlistFile = target.childFile('Info.plist');
+  await infoPlistFile.writeAsString('''
+  <?xml version="1.0" encoding="UTF-8"?>
+  <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+  <plist version="1.0">
+  <dict>
+    <key>CFBundleDevelopmentRegion</key>
+    <string>en</string>
+    <key>CFBundleExecutable</key>
+    <string>$name</string>
+    <key>CFBundleIdentifier</key>
+    <string>io.flutter.flutter.native_assets.$name</string>
+    <key>CFBundleInfoDictionaryVersion</key>
+    <string>6.0</string>
+    <key>CFBundleName</key>
+    <string>$name</string>
```

Comment:
I believe this must be max 15 characters though I don't know how it's enforced.
> This name can contain up to 15 characters. 

https://developer.apple.com/documentation/bundleresources/information_property_list/cfbundlename